test/csuite/random_directio/main.c

```diff
@@ -0,0 +1,1127 @@
+	buf1 = td->data;
+	buf2 = &td->data[kvsize];
+
+	/*
+	 * Continuing writing until we're killed.
+	 */
+	printf("Thread %" PRIu32 "\n", td->id);
+	for (i = 0; ; ++i) {
+	again:
+		retryable_error = false;
+		/*
+		if (i > 0 && i % 10000 == 0)
+			printf("Thread %d completed %d entries\n",
+			    (int)td->id, (int)i);
+		*/
+
+		gen_kv(buf1, kvsize, i, td->id, large, true);
+		gen_kv(buf2, kvsize, i, td->id, large, false);
+
+		testutil_check(session->begin_transaction(session, NULL));
+		cursor->set_key(cursor, buf1);
+		/*
+		 * Every 1000th record write a very large value that exceeds the
+		 * log buffer size.  This forces us to use the unbuffered path.
+		 */
+		if (i % 1000 == 0) {
+			cursor->set_value(cursor, large);
+		} else {
+			cursor->set_value(cursor, buf2);
+		}
+		testutil_check(cursor->insert(cursor));
+
+		/*
+		 * The reverse table has no very large records.
+		 */
+		rev->set_key(rev, buf2);
+		rev->set_value(rev, buf1);
+		testutil_check(rev->insert(rev));
+
+		/*
+		 * If we are doing a schema test, generate operations
+		 * for additional tables.  Each table has a 'lifetime'
+		 * of 4 values of the id.
+		 */
+		if (td->schema_test) {
+			if (!retryable_error && SCHEMA_OP(i, 0)) {
+				/* Create a table. */
+				gen_table_name(buf1, kvsize, i, td->id);
+				testutil_check(session->create(session, buf1,
+				    "key_format=S,value_format=S"));
+			}
+			if (!retryable_error && SCHEMA_OP(i, 1)) {
+				/* Insert a value into the table. */
+				gen_table_name(buf1, kvsize, i - 1, td->id);
+				testutil_check(session->open_cursor(session,
+				    buf1, NULL, NULL, &sch));
+				sch->set_key(sch, buf1);
+				sch->set_value(sch, buf1);
+				testutil_check(sch->insert(sch));
+				sch->close(sch);
+			}
+			if (!retryable_error && SCHEMA_OP(i, 2)) {
+				/* Rename the table. */
+				gen_table_name(buf1, kvsize, i - 2, td->id);
+				gen_table2_name(buf2, kvsize, i - 2, td->id);
+				/*
+				 * XXX
+				 * We notice occasional EBUSY errors from
+				 * rename, even though neither URI should be
+				 * used by any other thread.
+				 */
+				/*
+				printf(" rename(\"%s\", \"%s\")\n", buf1, buf2);
+				*/
+				if ((ret = session->rename(session, buf1, buf2,
+				    NULL)) == EBUSY) {
+					printf("rename(\"%s\", \"%s\") failed,"
+					    " retrying transaction\n", buf1,
+					    buf2);
+					retryable_error = true;
+				} else {
+					if (ret != 0)
+						printf("FAIL: "
+						    "rename(\"%s\", \"%s\") "
+						    "returns %d: %s\n",
+						    buf1, buf2, ret,
+						    wiredtiger_strerror(ret));
+					testutil_check(ret);
+				}
+			}
+			if (!retryable_error && SCHEMA_OP(i, 3)) {
+				/* Update the single value in the table. */
+				gen_table_name(buf1, kvsize, i - 3, td->id);
+				gen_table2_name(buf2, kvsize, i - 3, td->id);
+				testutil_check(session->open_cursor(session,
+				    buf2, NULL, NULL, &sch));
+				sch->set_key(sch, buf1);
+				sch->set_value(sch, buf2);
+				testutil_check(sch->insert(sch));
```

Comment:
Should this be `sch->update` instead of `sch->insert`? Or at least that's what the comment at line 401 implied to me.